library/bignum.c

```diff
@@ -54,6 +54,132 @@
 #define MPI_VALIDATE(cond)                                           \
     MBEDTLS_INTERNAL_VALIDATE(cond)
 
+/*
+ * Compare signed values in constant time
+ */
+int mbedtls_mpi_lt_mpi_ct(const mbedtls_mpi *X,
+                          const mbedtls_mpi *Y,
+                          unsigned *ret)
+{
+    mbedtls_ct_condition_t different_sign, X_is_negative, Y_is_negative, result;
+
+    MPI_VALIDATE_RET(X != NULL);
+    MPI_VALIDATE_RET(Y != NULL);
+    MPI_VALIDATE_RET(ret != NULL);
+
+    if (X->n != Y->n) {
+        return MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+    }
+
+    /*
+     * Set sign_N to 1 if N >= 0, 0 if N < 0.
```

Comment:
(minor; can be fixed in a later PR) presumably `sign_N` should be `N_is_negative`?

Reply:
and the values are swapped (the comment says the boolean is true for a non-negative N)

Reply:
Fixed in https://github.com/Mbed-TLS/mbedtls/pull/7590/commits/60d4b9f739d245f9be89a2e9d573d7ad89ff3c99